### PR TITLE
Users

### PR DESCRIPTION
**Pull request by qbaonguyen98**
Resolse: User APIs

Description: create APIs for get user list, get user information, update user by admin, update user profile and delete user

Testing:
user-apis

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,3 +1,4 @@
 JWT_SECRET=jwt_secret
 PORT=3000
```
Comment (NHTNhi) [inline .env.sample]: Change to port 5000

File: src/repositories/user-profile.repository.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  };
+
+  public find = async (
+    conditions: MongooseFilterQuery<UserProfileDocument> = {},
+    options: QueryOptions<UserProfileDocument> = {},
+  ): Promise<UserProfileDocument[]> => {
+    let userQuery = UserProfileModel.find(
+      {
+        ...conditions,
+        isDeleted: false,
```
Comment (NHTNhi) [inline src/repositories/user-profile.repository.ts]: Remove

File: src/services/user.service.ts [hunk 3]
```diff
@@ -1,10 +1,159 @@
...
+    }
+
+    const profile = await this.userProfileRepository.findOne({ _id: user.profileId });
+    profile.firstName = userData.firstName;
+    profile.lastName = userData.lastName;
+    profile.gender = userData.gender;
+    profile.dateOfBirth = userData.dateOfBirth;
+    await this.userProfileRepository.save(profile);
+  }
+
```
Comment (NHTNhi) [inline src/services/user.service.ts:109]: Replace by userProfileReposity.findOneAndUpdate() function

**Review by NHTNhi — APPROVED** [review]

**Review by h2312 — APPROVED** [review]
Code change P, Break test coverage P

**Pull request merged**